### PR TITLE
CORTX-28929 ceph/s3-tests runner

### PR DESCRIPTION
**Pull request by apoorva-rao**
# Problem Statement
- Implementation of ceph/s3-tests runner
- Basic Makefile to clone s3-tests repo, setup requirements and generate configuration file
- Execution details present in [CORTX-29322](https://jts.seagate.com/browse/CORTX-29322)

# Design
-  https://seagate-systems.atlassian.net/wiki/spaces/~509806131/pages/914882893/CORTX-28247+Integrating+test+result+reporting+of+ceph+s3-tests+with+cortx-test

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: commons/params.py [hunk 1]
```diff
@@ -125,3 +125,10 @@
 PROV_SKIP_TEST_FILES_HEALTH_CHECK_PREFIX = ['test_prov', 'test_failure_domain',
                                             'test_multiple_config_deploy', 'test_cont_deployment',
                                             "test_di_deployment"]
+
+# Ceph s3-tests Runner Params
+S3TESTS_DIR = "s3-tests"
+S3TESTS_CONF_ENV = "S3TEST_CONF"
+S3TESTS_CONF_FILE = "s3tests.conf"
+REPORTS_DIR = "reports"
+VIRTUALENV_DIR = "virtualenv"
```
Comment (seagate-sarang-sawant) [inline commons/params.py:134]: why would you need to specify virtualenv in Params?  It is responsibility of another script to take care setting the py env and pythonpath.

File: scripts/ceph_s3_tests/Makefile [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+# This file will setup & configure the ceph/s3-tests repo.
+# Commands:
+#	1. make help # To check supported tools by this script.
+#	2. make setup-s3tests # Clone and setup s3-tests repo under cortx-test
+#	3. make configure-s3tests  # Generate configuration file for running tests
+# Note: Please provide existing s3account user access, secret keys through command line.
+
+CORTX_TESTS_DIR=/root/cortx-test
```
Comment (seagate-sarang-sawant) [inline scripts/ceph_s3_tests/Makefile:25]: Let user provide this path as well.

File: scripts/ceph_s3_tests/Makefile [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ENDPOINT=s3.seagate.com
```
Comment (seagate-sarang-sawant) [inline scripts/ceph_s3_tests/Makefile:26]: This endpoint indicates if you are using /etc/hosts. Is it ?

File: commons/params.py (hunk 1 above)
Comment (apoorva-rao) [reply commons/params.py:134]: Using this to get the path to nosetests command. The Makefile executes the ceph/s3-tests bootstrap script which takes care of setting this virtual environment up

File: scripts/ceph_s3_tests/Makefile (hunk 2 above)
Comment (apoorva-rao) [reply scripts/ceph_s3_tests/Makefile:25]: Yes, user can override this by specifying it at the command line. Need to get the relative directory for cloning the repo, which is why I have specified a default value

File: scripts/ceph_s3_tests/Makefile (hunk 3 above)
Comment (apoorva-rao) [reply scripts/ceph_s3_tests/Makefile:26]: Specifying a default value here, but expectation is for user to specify ENDPOINT at command line if s3.seagate.com cannot be resolved

File: scripts/ceph_s3_tests/Makefile (hunk 2 above)
Comment (seagate-sarang-sawant) [reply scripts/ceph_s3_tests/Makefile:25]: ok

File: commons/params.py (hunk 1 above)
Comment (seagate-sarang-sawant) [reply commons/params.py:134]: Let it assume project root to setup virtual env .  Also does it setup virenv every time. There must be an option ruse the older virenv as well.

**Review by seagate-sarang-sawant — APPROVED** [review]

File: commons/params.py (hunk 1 above)
Comment (apoorva-rao) [reply commons/params.py:134]: Yes, agreed but here we cleanup the repo for each execution so this s3-tests repo clone is temporary in the local workspace.

**Review by swapnil-khandare — APPROVED** [review]
Approving for merge

**Pull request merged**